### PR TITLE
Feature/proxy improvement

### PR DESCRIPTION
**Pull request by rodrigopavezi**
It will replace the httpFetch to use got module
Changed some code that was referencing response.status to response.statusCode
Updated some unit tests
Updated proxy integration tests and add one test to check if when using net module it will work with a session proxy setup

Installer and player tested locally.

@tejohnso @stulees please review. cheers.

### CODE REVIEW

File: installer/gcs.js [hunk 1]
```diff
@@ -90,9 +90,9 @@ module.exports = {
 
       return network.httpFetch(url)
       .then((resp)=>{
-        if (resp.status === 304) {return {local: useLocalData ? getLocalContent(gcsPath, localGCSData) : null};}
-        if (resp.status !== 200) {throw new Error(`${resp.status} ${resp.statusText}`);}
-        return {remote: resp.json(), generation: resp.headers.get("x-goog-generation")};
+        if (resp.statusCode === 304) {return {local: useLocalData ? getLocalContent(gcsPath, localGCSData) : null};}
+        if (resp.statusCode !== 200) {throw new Error(`${resp.status} ${resp.statusText}`);}
+        return {remote: resp.json(), generation: resp.headers["x-goog-generation"]};
```
Comment (tejohnso) [inline installer/gcs.js]: Why are these changes required?  Does `got` return a different object for `resp.headers`?

Comment (rodrigopavezi) [inline installer/gcs.js]: @tejohnso it returns an array instead of an object with a get method. cheers

**Comment by rodrigopavezi** [comment]
@tejohnso please review latest commit. cheers

**Comment by rodrigopavezi** [comment]
@tejohnso please review last commit. It has the changes to resolve localhost.
